src/Serilog/Parsing/MessageTemplateParser.cs

```diff
@@ -86,9 +86,8 @@ static MessageTemplateToken ParsePropertyToken(int startAt, string messageTempla
             next = startAt + 1;
 
             var rawText = messageTemplate.Substring(first, next - first);
-            var tagContent = messageTemplate.Substring(first + 1, next - (first + 2));
-            if (tagContent.Length == 0 ||
-                !IsValidInPropertyTag(tagContent[0]))
+            var tagContent = rawText.Substring(1, next - (first + 2));
```

Comment:
reading from `rawText` instead of `messageTemplate` makes the intent clearer and allows to make the `startIndex` argument const to `1` (skipping the `{` in `rawText`)